d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -626,6 +644,41 @@ private void emitSDStatusUpdateReceiptEvents(Map<String, XdsAndD2Uris> updates,
     }
   }
 
+
+  private class UriPropertiesCollectionResourceWatcher implements XdsClient.D2URICollectionResourceWatcher
+  {
+    private final UriPropertiesResourceWatcher _delegate;
+
+    UriPropertiesCollectionResourceWatcher(String clusterName)
+    {
+      _delegate = new UriPropertiesResourceWatcher(clusterName);
+    }
+
+    @Override
+    public void onError(Status error)
+    {
+      _delegate.onError(error);
+    }
+
+    @Override
+    public void onReconnect()
+    {
+      _delegate.onReconnect();
+    }
+
+    @Override
+    public void onChanged(XdsClient.D2URICollectionUpdate update)
```

Comment:
when the update is empty (new D2URICollectionUpdate()), this method should call `_delegate.onChanged(new XdsClient.D2URIMapUpdate(null));`. Otherwise, the empty map will be published and wipe client side existing data. 

Please add unit tests for this method to cover both normal data and empty data scenarios.